src/Documentation/RuleSetDocumentationGenerator.php

```diff
@@ -46,8 +47,52 @@ public function generateRuleSetsDocumentation(RuleSetDescriptionInterface $defin
         $titleLine = str_repeat('=', \strlen($title));
         $doc = "{$titleLine}\n{$title}\n{$titleLine}\n\n".$definition->getDescription();
 
+        $warnings = [];
+        if ($definition instanceof DeprecatedRuleSetDescriptionInterface) {
+            $deprecationDescription = <<<'RST'
+
+                This rule set is deprecated and will be removed in next major version
+                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
@Wirone 

Probably needs a slight adjustment following #7332?

```suggestion
                This rule set is deprecated and will be removed in the next major version
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```